### DISCUSSION TITLE
Squeeze film damper element

### DISCUSSION THREAD

**Discussion by wigging** [petrobras/ross#895] [Q&A] [answered]
How would I represent a squeeze film damper as part of the rotor model? Would it be implemented as a `BearingFluidFlow`? I read example 7 in the documentation which is for hydrodynamic bearings. But I'm not sure if that is the same thing as a squeeze film damper.

**Comment by raphaeltimbo** [comment, marked as answer]
Hi @wigging. 
Currently ROSS does not have the capability to calculate the coefficients for a squeeze film damper.
But if you have the coefficients from some documentation or from another software you can include the sfd using a point mass and adding it simillar to a support in the model. 
The tutorial has an example for that, with `bearing0` being added as a support:
![image](https://user-images.githubusercontent.com/18506378/182381092-2af791a9-c6cd-4f1e-b693-2a58e265073c.png)

**Reply by wigging** [reply]
Are you referring to the example in section 6.1 of the "Tutorial - Modeling" documentation?

I am using a squeeze film damper model that calculates the film pressure along the circumferential and axial directions. The model also provides the inertia force coefficients (Mxx, Mxy, Myy, Myx, etc.) and the damping force coefficients (Cxx, Cxy, Cyy, Cyx, etc.). In the example you shared above, how would I use the inertia coefficients and pressures calculated from my damper model?

**Reply by wigging** [reply]
@raphaeltimbo In addition to my previous comment, can you elaborate on what the `kxx` and `kyy` inputs represent? I understand that they are the stiffness in the x and y directions with units of N/m. But what does that stiffness represent. Is it the stiffness of the bearing, stiffness of the rotor, or the stiffness of the housing that contains the entire rotor assembly?

**Reply by raphaeltimbo** [reply]
> Are you referring to the example in section 6.1 of the "Tutorial - Modeling" documentation?

Yes.

> 
> I am using a squeeze film damper model that calculates the film pressure along the circumferential and axial directions. The model also provides the inertia force coefficients (Mxx, Mxy, Myy, Myx, etc.) and the damping force coefficients (Cxx, Cxy, Cyy, Cyx, etc.). In the example you shared above, how would I use the inertia coefficients and pressures calculated from my damper model?

Taking the tutorial example as reference, the damping coefficients would be added to the ‘bearing0’ and the inertia coefficients to the point mass element.
But I have just noticed that the PointMass class does not have arguments for the cross-coupled inertia coefficients – Mxy and Myx. 
Later I will add this to the code. But if this is urgent to you and if in your model those are not 0, you could add them to the point mass element by modifying you ROSS source code. 
Basically, you would have to add the mxy and myx to the `__init__` arguments and change the following M() method to assemble the correct matrix:

https://github.com/ross-rotordynamics/ross/blob/f9a725918de703891f5608289621399f4a3c8472/ross/point_mass.py#L138-L162

> @raphaeltimbo In addition to my previous comment, can you elaborate on what the kxx and kyy inputs represent? I understand that they are the stiffness in the x and y directions with units of N/m. But what does that stiffness represent. Is it the stiffness of the bearing, stiffness of the rotor, or the stiffness of the housing that contains the entire rotor assembly?

In the tutorial example 'bearing0' could represent the bearing housing. So kxx and kyy is the stiffness for that component. But notice that you can go on adding point masses and new 'bearing' elements to represent other components.
For example, this is the model for a generator in which we added the bearing, the bearing house and the support, each with their respective stiffness:

![image](https://user-images.githubusercontent.com/18506378/182710344-617a7f6a-59ee-4c08-9691-e50b88de72ff.png)

**Reply by wigging** [reply]
Ok, I'll see if I can use the `PointMass` approach. Also, `PointMass` is not documented in the API reference.
